boxsdk/object/base_object.py

```diff
@@ -155,8 +155,24 @@ def delete(self, params=None, headers=None):
         return box_response.ok
 
     def __eq__(self, other):
-        """Equality as determined by object id"""
-        return self._object_id == other.object_id
+        """Equality as determined by object id and type"""
+        if other is None:
```

Comment:
None is valid to pass to `isinstance()`, so this check is redundant with the next check and is unnecessary.